query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/converters/DefaultColumnRequestConverter.java

```diff
@@ -207,6 +211,70 @@ private boolean handleConversionForNullOrEmptyBytesLiteral(
     return false;
   }
 
+  private boolean isFilterForArrayColumnType(Filter filter, ColumnRequestContext context) {
+    return isSimpleAttributeExpression(filter.getLhs())
+        && filter.getRhs().getValueCase().equals(LITERAL)
+        && context.isArrayColumnType();
+  }
+
+  /** Handles value conversion of a array expression based */
+  private void handleConversionForArrayColumnExpression(
+      String lhs, String operator, Expression rhs, StringBuilder builder, Builder paramsBuilder) {
+    Value value = rhs.getLiteral().getValue();
+
+    if (handleConversionForNullOrEmptyArrayLiteral(lhs, operator, builder, value)) return;
+
+    // support only equals and IN operator
+    // both of them are handled as contains check to align with existing implementation
+    if (operator.equals("=")
+        || operator.equals("IN")
+        || operator.equals("!=")
+        || operator.equals("NOT IN")) {
```

Comment:
In the same, backlog ticket, also add that other operators will be supported when required.